packages/web/components/place-limit-tool/limit-price-selector.tsx

```diff
@@ -45,32 +47,54 @@ export const LimitPriceSelector: FC<LimitPriceSelectorProps> = ({
         : InputMode.Percentage
     );
 
+    if (inputMode === InputMode.Price) {
+      priceState.setPriceLock(false);
+    } else {
+      priceState.setPriceLock(true);
+    }
+
+    if (priceState.isBeyondOppositePrice || !priceState.manualPercentAdjusted) {
+      priceState.setPercentAdjusted("0");
+    }
+
     if (input) input.focus();
-  }, [inputMode, input]);
+  }, [inputMode, input, priceState]);
 
-  // Adjust the percentage adjusted when the spot price changes and user is inputting a price
+  // Adjust order price as spot price changes until user inputs a price
   useEffect(() => {
-    if (
-      priceState.spotPrice &&
-      priceState.orderPrice.length > 0 &&
-      inputMode === InputMode.Price
-    ) {
-      const manualPrice = new Dec(priceState.orderPrice);
-      const percentAdjusted = manualPrice
-        .quo(priceState.spotPrice)
-        .sub(new Dec(1))
-        .mul(new Dec(100));
+    if (inputMode === InputMode.Price && !priceState.priceLocked) {
+      const formattedPrice = formatPretty(
+        priceState.priceFiat,
+        getPriceExtendedFormatOptions(priceState.priceFiat.toDec())
+      )
+        .replace("$", "")
+        .replace(/,/g, "");
+      priceState._setPriceUnsafe(formattedPrice);
+      priceState._setPercentAdjustedUnsafe("0");
+    }
 
-      priceState._setPercentAdjustedUnsafe(
-        percentAdjusted.isZero() || manualPrice.isZero()
-          ? ""
-          : formatPretty(percentAdjusted.abs(), {
-              maxDecimals: 3,
-            }).toString()
+    if (inputMode === InputMode.Percentage) {
+      console.log(
```

Comment:
Do we need it for prod?

Reply:
Good spot, was using it to ensure the price was updating (it can take a while)